### PR TITLE
Fixes #14222 - Hide Non Displayable Repos from CDN repos

### PR DESCRIPTION
**Pull request by parthaa**
CDN doesn't enforce Docker entitlements, so no point showing it in the UI

### CODE REVIEW

**Comment by parthaa** [comment]
Change documented in http://imgur.com/7kfl0s4

**Comment by jlsherrill** [comment]
Won't they still show up under 'other' with this change?

**Comment by parthaa** [comment]
> Won't they still show up under 'other' with this change?

That is correct . I could not find the place where you hide things ...

**Comment by jlsherrill** [comment]
@parthaa what is being hidden currently are 'repositories', but you actually want to hide the repository sets themselves.

I'd change that code to just do something like:

```
          if prod_content.content.type == ::Katello::Repository::CANDLEPIN_DOCKER_TYPE
            key = :skip

<SNIP>

          unless key == :skip
                      tabs[key][:products][product.id] ||= []
                      tabs[key][:products][product.id] << prod_content
           end
```

**Comment by ehelms** [comment]
An API update to also hide them would be nice

**Comment by parthaa** [comment]
> An API update to also hide them would be nice

@ehelms I thought we were ok letting em set from API if they knew the exact values ? .. No ?

**Comment by ehelms** [comment]
Yes - but we need to hide the output from API requests.

**Comment by parthaa** [comment]
Added logic to hide Docker Manifest and Ostree Repos (if ostree content type is not enabled)

**Comment by parthaa** [comment]
Other changes to check thanks to suggestions by @ehelms  and @jlsherrill  . API repository-set call wont show any docker entitlements
<pre>
$ hammer repository-set list --organization-id=1 --product="Red Hat Container Images"
---|------|-----
ID | TYPE | NAME
---|------|-----
</pre>

With OSTree
<pre>
$ hammer repository-set list --organization-id=1 --product="Red Hat Enterprise Linux Atomic Host"
-----|--------|---------------------------------------------------
ID   | TYPE   | NAME                                              
-----|--------|---------------------------------------------------
3880 | yum    | Red Hat Enterprise Linux Atomic Host (Debug RPMs) 
3822 | ostree | Red Hat Enterprise Linux Atomic Host (Trees)      
3830 | yum    | Red Hat Enterprise Linux Atomic Host (RPMs)       
3824 | yum    | Red Hat Enterprise Linux Atomic Host (Source RPMs)
3823 | file   | Red Hat Enterprise Linux Atomic Host (Images)     
-----|--------|---------------------------------------------------
</pre>

and if you dont have ostree enabled (to test you can remove ostree.rb from /katello/lib/katello/repository_types/ostree.rb  completely optional...)

Without OSTree
<pre>
$ hammer repository-set list --organization-id=1 --product="Red Hat Enterprise Linux Atomic Host"
-----|--------|---------------------------------------------------
ID   | TYPE   | NAME                                              
-----|--------|---------------------------------------------------
3880 | yum    | Red Hat Enterprise Linux Atomic Host (Debug RPMs) 
3822 | ostree | Red Hat Enterprise Linux Atomic Host (Trees)      
3830 | yum    | Red Hat Enterprise Linux Atomic Host (RPMs)       
3824 | yum    | Red Hat Enterprise Linux Atomic Host (Source RPMs)
3823 | file   | Red Hat Enterprise Linux Atomic Host (Images)     
-----|--------|---------------------------------------------------
</pre>

File: app/models/katello/candlepin/product_content.rb [hunk 1]
```diff
@@ -34,5 +34,15 @@ def content_override(activation_key)
...
     end
+
+    def content_type
+      self.content.type
+    end
+
+    def displayable?
+      return true  unless (content_type == ::Katello::Repository::CANDLEPIN_DOCKER_TYPE) ||
+                          (::Katello::RepositoryTypeManager.find(Repository::OSTREE_TYPE).nil? &&
+                            content_type == ::Katello::Repository::CANDLEPIN_OSTREE_TYPE)
```
Comment (ehelms) [inline app/models/katello/candlepin/product_content.rb]: This could just return the value of this expression since it results in a boolean.

**Comment by parthaa** [comment]
@ehelms  updated

**Comment by ehelms** [comment]
ACK

**Pull request merged**